Refactor web authentication

krzysdabro opened the pull request:
- Replace regex with HTML parsers
- Move doWebAuth to VWWebSession
- Split doWebAuth into multiple methods
- Add Terms and Conditions form handling

tillsteinbach commented:
Thank you for your contribution. This looks great! I have one thing where I'm unsure. I was also thinking in the past of automatically accepting terms and conditions and decided not to do it and ask the user to do it himself. The reason is that the user otherwise accepts something he will definitely not have looked at. While we all know that anyway no one will read the terms I'm fearing that just automatically accepting Terms and Conditions could be a reason for VW to be upset and to lock us out of the API.

tillsteinbach commented:
I checked out the branch, but I get this error right away:
`2024-12-19T18:47:40+0100:CRITICAL:There was a problem when communicating with WeConnect. If this problem persists please open a bug report: Retrieving credentials page was not successful, status code: 400`

krzysdabro commented:
It looks like double slash in URL is a no-go. I've updated the solution to use `urljoin`. Additionally, I’ve added a switch for accepting terms, which is set to `False` by default to maintain the previous behavior.

In my case, handling the terms is necessary because VW somehow can't save that I have accepted them and prompts me every time I log in through the library.

tillsteinbach commented:
> I checked out the branch, but I get this error right away: `2024-12-19T18:47:40+0100:CRITICAL:There was a problem when communicating with WeConnect. If this problem persists please open a bug report: Retrieving credentials page was not successful, status code: 400`

This error is gone now for me.

krzysdabro commented:
> > In my case, handling the terms is necessary because VW somehow can't save that I have accepted them and prompts me every time I log in through the library.
> 
> Also if you comment out this: [7a0eafc](https://github.com/tillsteinbach/WeConnect-python/commit/7a0eafc33ec82fe04d9ad3e7abd32ab1966733f0)

I have already rebased the branch and this is gone. I think the PR is ready.
